### DISCUSSION TITLE
Does rule engine support Expression in Value field

### DISCUSSION THREAD

**Discussion by dtiwarATS** [CacheControl/json-rules-engine#249] [General]
Hi There,

I have a use case where the value field has an expression like this: 

{
   "fact": "fact1",
   "operator": "equal",
   "value": "100 - {var1} - {var2} - {var2}"
}

**Note:-** Var1, Var2, etc are nothing but a fact.

Up to the best of my knowledge, I guess the facts comparison allows to compare only one fact with another but I was curious to know if there is a way to support the above conditions.

Thanks.

**Comment by CacheControl** [comment]
Hi @dtiwarATS, yes that should be achievable using [fact comparison](https://github.com/CacheControl/json-rules-engine/blob/master/docs/rules.md#comparing-facts) - see [example](https://github.com/CacheControl/json-rules-engine/blob/master/examples/08-fact-comparison.js). One approach would be to have a [dynamic fact](https://github.com/CacheControl/json-rules-engine/blob/master/examples/03-dynamic-facts.js) that represents the expression you posted, which would [resolve](https://github.com/CacheControl/json-rules-engine/blob/master/docs/almanac.md#almanacfactvaluefact-fact-object-params-string-path---promise) the fact values in code and evaluate the expression. 

Additionally, you might also look at using fact params to provide a data structure describing the expression, which could then be parsed by the dynamic fact. This would avoid hard coding each expression within its own fact.
